airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,105 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Transfers data from Snowflake into a S3 Bucket."""
+from typing import List, Optional
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+
+
+class SnowflakeToS3Operator(BaseOperator):
+    """
+    UNLOAD a query or a table from Snowflake to S3
+
+    :param schema: reference to a specific schema in snowflake database
+    :type schema: str
+    :param table: reference to a specific table in snowflake database
+    :type table: str
+    :param s3_bucket: reference to a specific S3 bucket where the data will be saved
+    :type s3_bucket: str
+    :param s3_key: reference to a specific S3 key within the previous bucket
+    :type s3_key: str
+    :param file_format: can be either a previous file format created in Snowflake console
+        or hardcoded one like ``type = csv field_delimiter = ',' skip_header = 1``
+    :type file_format: str
+    :param sql: optional parameter to unload a customized query instead an entire table
+    :type sql: str
+    :param snowflake_conn_id: reference to a specific snowflake database
+    :type snowflake_conn_id: str
+    :param unload_options: reference to a list of UNLOAD options (SINGLE, MAX_FILE_SIZE,
+        OVERWRITE etc). Each element of the list has to be a string
+    :type unload_options: list
+    """
+
+    template_fields = (
+        's3_key',
+        's3_bucket',
+        'sql',
+        'table',
+        'snowflake_conn_id',
+    )
+    template_ext = ('.sql',)
+    ui_color = '#ffebb2'
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        s3_bucket: str,
+        s3_key: str,
+        file_format: str,
+        schema: Optional[str] = None,
+        table: Optional[str] = None,
+        sql: Optional[str] = None,
```

Comment:
```suggestion
        table_or_query: Optional[str] = None,
```
I would suggest to keep it rather simple and allow passing table or query to one variable and just pass it to the `COPY` string. Otherwise you would have to do more validation on this - so that either schema and table or a query was passed.

Ref: https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#syntax

Reply:
Done it.